include/ginkgo/core/distributed/base.hpp

```diff
@@ -67,6 +67,21 @@ class DistributedBase {
 };
 
 
+/**
+ * DistributedLocalSize is a feature class providing `get_local_size` which
+ * return the size of local operator.
+ */
+class DistributedLocalSize {
```

Comment:
I was also thinking about putting it into the `DistributedBase`, but I wonder it is out of the scope because it only handles the communicator now.
`non_local_size` can not go into this base because the vector does not have the idea about non_local_size.
Or, we just put non_local_size of vector is dim<2>(0, 0).
I do not have an idea about the usage of non_local_size out of the matrix

Reply:
No, I only mean the functionality but don't mean put it into the DistributedLocalSize.
I agree with user may expect something similar to `get_local_size()` for `non_local_matrix()`.
For me, if we need to get_non_local_size(), we need to put `get_local_size()` into `DistributedLocalSize` and `get_non_local_size()` into `DistributedNonLocalSize` because vector does not have non_local information such that we can not put them together into `DistrbutedBase`. One in `DistrbutedBase` and another in `DistributedNonLocalSize` are also inconsistent in my mind. (unless we put non_local_size = 0 in vector cases)

Reply:
I think it should be okay to have only local_size in DistributedBase. A separate interface class is too clunky IMO. But given that this probably will be overwritten, I dont have a strong opinion, and we can discuss this later.

Reply:
I move it to DistributedBase now.